CONTRIBUTING.md

```diff
@@ -0,0 +1,184 @@
+
+## Detailed Workflow
+
+### Fork the repo
+
+Use github’s interface to make a fork of the repo, then add that repo as an upstream remote:
+
+```
+git remote add upstream https://github.com/the-ninja-turtles/turtle
+```
+
+### Cut a feature branch from develop
+
+These commands will help you do this:
+
+``` bash
+# Creates your branch and takes you there
+git checkout -b `your-branch-name` develop
+```
+
+### Make commits to your feature branch.
+
+Prefix each commit like so
+  - (feat) Added a new feature
+  - (fix) Fixed inconsistent tests
+  - (refactor) ...
+  - (test) ...
+  - (doc) ...
+  - (build) ...
+
+Make changes and commits on your branch, and make sure that you
+only make changes that are relevant to this branch. If you find
+yourself making unrelated changes, make a new branch for those
+changes.
+
+#### Commit Message Guidelines
+
+- Commit messages should be written in the present tense; e.g. "Fix continuous
+  integration script".
+- The first line of your commit message should be a brief summary of what the
+  commit changes. Aim for about 70 characters max. Remember: This is a summary,
+  not a detailed description of everything that changed.
+- If you want to explain the commit in more depth, following the first line should
+  be a blank line and then a more detailed description of the commit. This can be
+  as detailed as you want, so dig into details here and keep the first line short.
+
+### Rebase upstream changes into your branch
+
+Once you are done making changes, you can begin the process of getting
+your code merged into the main repo. Step 1 is to rebase upstream
+changes to the master branch into yours by running this command
+from your branch:
+
+```bash
+git pull --rebase upstream develop
+```
+
+This will start the rebase process. You must commit all of your changes
+before doing this. If there are no conflicts, this should just roll all
+of your changes back on top of the changes from upstream, leading to a
+nice, clean, linear commit history.
+
+If there are conflicting changes, git will start yelling at you part way
+through the rebasing process. Git will pause rebasing to allow you to sort
+out the conflicts. You do this the same way you solve merge conflicts,
+by checking all of the files git says have been changed in both histories
+and picking the versions you want. Be aware that these changes will show
+up in your pull request, so try and incorporate upstream changes as much
+as possible.
+
+You pick a file by `git add`ing it - you do not make commits during a
+rebase.
+
+Once you are done fixing conflicts for a specific commit, run:
+
+```bash
+git rebase --continue
+```
+
+This will continue the rebasing process. Once you are done fixing all
+conflicts you should run the existing tests to make sure you didn’t break
+anything, then run your new tests (there are new tests, right?) and
```

Comment:
> you should run the existing tests ... then run your new tests

Are you implying that there should somehow be 2 test runs? First time you only run old tests (how?) and then — only new tests (how?)?
Wouldn't it be easier to just run all tests you have at this point to make sure nothing breaks?